### PR TITLE
Feature/add tests for home presenter

### PR DESCRIPTION
**Pull request by jaisonv**
Did some refactoring and added tests for HomePresenter.

### CODE REVIEW

File: CocktailRecipes/CocktailRecipesTests/Features/Home/HomePresenterTests.swift [hunk 1]
```diff
@@ -0,0 +1,56 @@
...
+
+    private func makeNetworkManager() -> NetworkManager {
+
+        let data = makeDrinkArrayData()
+
+        let session = MockSession(response: (data, URLResponse: nil, error: nil))
+        return NetworkManager(session: session)
+    }
+
+    private func makeDrinkArrayData() -> Data {
```
Comment (EricRado) [inline CocktailRecipes/CocktailRecipesTests/Features/Home/HomePresenterTests.swift]: I prefer `contstructDrinkData()`

File: CocktailRecipes/CocktailRecipesTests/Features/Home/HomePresenterTests.swift [hunk 2]
```diff
@@ -0,0 +1,56 @@
...
+    private func makeSUT() -> HomePresenter {
+        return HomePresenter(networkManager: makeNetworkManager())
+    }
+
+    private func makeFailingSUT() -> HomePresenter {
+        let session = MockSession(response: (Data(), URLResponse: nil, error: nil))
+        return HomePresenter(networkManager: NetworkManager(session: session))
+    }
+
+    private func makeNetworkManager() -> NetworkManager {
```
Comment (EricRado) [inline CocktailRecipes/CocktailRecipesTests/Features/Home/HomePresenterTests.swift]: here `constructNetworkManager`

File: CocktailRecipes/CocktailRecipesTests/Features/Home/HomePresenterTests.swift [hunk 3]
```diff
@@ -0,0 +1,56 @@
...
+        XCTAssertEqual(sut.dataSource(for: .random).count, 0)
+        XCTAssertEqual(sut.dataSource(for: .latest).count, 0)
+        XCTAssertEqual(sut.dataSource(for: .popular).count, 0)
+    }
+
+    private func makeSUT() -> HomePresenter {
+        return HomePresenter(networkManager: makeNetworkManager())
+    }
+
+    private func makeFailingSUT() -> HomePresenter {
```
Comment (EricRado) [inline CocktailRecipes/CocktailRecipesTests/Features/Home/HomePresenterTests.swift]: `forceSystemUnderTestFailure`

File: CocktailRecipes/CocktailRecipesTests/Features/Home/HomePresenterTests.swift [hunk 4]
```diff
@@ -0,0 +1,56 @@
...
+
+    func testDataSourceFailFetch() {
+        let sut = makeFailingSUT()
+
+        XCTAssertEqual(sut.dataSource(for: .random).count, 0)
+        XCTAssertEqual(sut.dataSource(for: .latest).count, 0)
+        XCTAssertEqual(sut.dataSource(for: .popular).count, 0)
+    }
+
+    private func makeSUT() -> HomePresenter {
```
Comment (EricRado) [inline CocktailRecipes/CocktailRecipesTests/Features/Home/HomePresenterTests.swift]: `constructSystemUnderTest`

File: CocktailRecipes/CocktailRecipesTests/Features/Home/HomePresenterTests.swift (hunk 1 above)
Comment (jaisonv) [reply CocktailRecipes/CocktailRecipesTests/Features/Home/HomePresenterTests.swift]: Changed.

File: CocktailRecipes/CocktailRecipesTests/Features/Home/HomePresenterTests.swift (hunk 2 above)
Comment (jaisonv) [reply CocktailRecipes/CocktailRecipesTests/Features/Home/HomePresenterTests.swift]: Changed.

File: CocktailRecipes/CocktailRecipesTests/Features/Home/HomePresenterTests.swift (hunk 3 above)
Comment (jaisonv) [reply CocktailRecipes/CocktailRecipesTests/Features/Home/HomePresenterTests.swift]: Changed to ```constructFailingHomePresenter```.

File: CocktailRecipes/CocktailRecipesTests/Features/Home/HomePresenterTests.swift (hunk 4 above)
Comment (jaisonv) [reply CocktailRecipes/CocktailRecipesTests/Features/Home/HomePresenterTests.swift]: Changed to constructHomePresenter.

**Pull request merged**